src/wp-includes/option.php

```diff
@@ -2144,20 +2144,47 @@ function update_network_option( $network_id, $option, $value ) {
 	 */
 	$value = apply_filters( "pre_update_site_option_{$option}", $value, $old_value, $option, $network_id );
 
+	$is_multisite = is_multisite();
+
+	/*
+	 * To get the actual raw old value from the database, any existing pre filters need to be temporarily disabled.
+	 * Immediately after getting the raw value, they are reinstated.
+	 * The raw value is only used to determine whether a value is present in the database. It is not used anywhere
+	 * else, and is not passed to any of the hooks either.
+	 */
+	if ( has_filter( "pre_site_option_{$option}" ) ) {
```

Comment:
What about the `pre_option_{$option}` filters? If this function is called when `! is_multisite()`, it will retrieve a regular option right below, via `get_network_option()`, which then calls `get_option()`.

If the option has a pre filter, we will not remove it here, which could lead to the same bug for why we introduced similar logic in `update_option()`. I think we may need to integrate it here too, even though that makes for clunky logic :/

Reply:
Something I was thinking about while reviewing this (and remembering the similar pattern in `get_option()` is whether there should be a way to get the unfiltered value without juggling hook keys like this. For example, we could add a parameter to these function that would bypass filters, if `true`, but default it to `false`. Alternately, we could add function like `get_option_raw()` which are used by `get_option()` but returns the unfiltered value.

Reply:
To be clear. I only think we should consider the approaches I mentioned above to refactor the API to get unfiltered options if there are benefits outside this specific use case. I would also consider them an enhancement for a future release, not something blocking for this PR.